odc/stats/plugins/mangroves.py

```diff
@@ -0,0 +1,113 @@
+"""
+Fractional Cover Percentiles
+"""
+from functools import partial
+from itertools import product
+from typing import Optional, Tuple
+
+import numpy as np
+import xarray as xr
+import dask
+from odc.algo import keep_good_only, erase_bad
+from odc.algo._masking import _fuse_mean_np, _fuse_or_np, _or_fuser, _xr_fuse
+from odc.algo._percentile import xr_quantile_bands
+from osgeo import gdal, ogr, osr
+
+from ._registry import StatsPluginInterface, register
+
+NODATA = 255
+
+
+class Mangroves(StatsPluginInterface):
+
+    NAME = "mangroves"
+    SHORT_NAME = NAME
+    VERSION = "0.0.1"
+    PRODUCT_FAMILY = "mangroves"
+
+    def __init__(
+        self,
+        pv_thresholds=[14, 38, 60],
+        tcw_threshold=-1850,
+        **kwargs,
+    ):
+        super().__init__(input_bands=["pv_pc_10", "qa", "wet_pc_10"], **kwargs)
+
+        self.mangroves_extent = kwargs.get('mangroves_extent', None)
+        self.pv_thresholds = pv_thresholds
+        self.tcw_threshold = tcw_threshold
+
+    @property
+    def measurements(self) -> Tuple[str, ...]:
+        _measurements = [
+            "canopy_cover_class"
+        ]
+        return _measurements
+    
+    def rasterize_mangroves_extent(shape_file, array_shape, orig_coords, resolution=(30, -30)):
+        source_ds = ogr.Open(shape_file)
+        source_layer = source_ds.GetLayer()
+
+        yt, xt = array_shape[1:]
+        xres, yres = resolution 
+        no_data = 0
+        xcoord, ycoord = orig_coords
+
+        geotransform = (xcoord - (xres * 0.5), xres, 0, ycoord - (yres * 0.5), 0, yres)
+
+        target_ds = gdal.GetDriverByName("MEM").Create("", xt, yt, gdal.GDT_Byte)
+        target_ds.SetGeoTransform(geotransform)
+        albers = osr.SpatialReference()
+        albers.ImportFromEPSG(3577)
+        target_ds.SetProjection(albers.ExportToWkt())
+        band = target_ds.GetRasterBand(1)
+        band.SetNoDataValue(no_data)
+
+        gdal.RasterizeLayer(target_ds, [1], source_layer, burn_values=[1])
+        return dask.array.from_array(band.ReadAsArray().reshape(array_shape), name=False)
+
+    def native_transform(self, xx):
+        """
+            no native transform required since
+            no reprojection in mangroves
+            return loaded data
+        """
+        return xx
```

Comment:
This is OK, but unnecessary as it identical to the default implementation is provided by the base class.

Reply:
deleted